waspc/e2e-test/GoldenTest.hs

```diff
@@ -0,0 +1,85 @@
+module GoldenTest
+  ( runGoldenTest,
+    makeGoldenTest,
+    GoldenTest,
+  )
+where
+
+import Common (getTestOutputsDir)
+import Control.Monad (filterM)
+import Data.List (sort)
+import Data.Text (pack, replace, unpack)
+import ShellCommands (ShellCommand, ShellCommandBuilder, ShellCommandContext (..), combineShellCommands, runShellCommandBuilder)
+import qualified StrongPath as SP
+import System.Directory (doesFileExist)
+import System.Directory.Recursive (getDirFiltered)
+import System.FilePath (makeRelative, takeFileName)
+import qualified System.FilePath as FP
+import System.Process (callCommand)
+import Test.Tasty (TestTree, testGroup)
+import Test.Tasty.Golden (goldenVsFileDiff)
+
+data GoldenTest = GoldenTest {_goldenTestName :: String, _goldenTestCommands :: ShellCommandBuilder [ShellCommand]}
+
+makeGoldenTest :: String -> ShellCommandBuilder [ShellCommand] -> GoldenTest
+makeGoldenTest name commands = GoldenTest {_goldenTestName = name, _goldenTestCommands = commands}
+
+-- | This runs a golden test by creating a Wasp project (via `wasp-cli new`), running commands,
+-- and then comparing all file outputs to the corresponding golden test output directory.
+runGoldenTest :: GoldenTest -> IO TestTree
+runGoldenTest goldenTest = do
+  testOutputsDirAbsSp <- getTestOutputsDir
+  let testOutputsDirAbsFp = SP.fromAbsDir testOutputsDirAbsSp
+  let currentOutputDirAbsFp = testOutputsDirAbsFp FP.</> (_goldenTestName goldenTest ++ "-current")
+  let goldenOutputDirAbsFp = testOutputsDirAbsFp FP.</> (_goldenTestName goldenTest ++ "-golden")
+
+  -- Remove existing current output files from a prior test run.
+  callCommand $ "rm -rf " ++ currentOutputDirAbsFp
```

Comment:
I hope this can never get a handle to my home directory...

Reply:
Strong path and prior checks for `waspc` have your back ;) We unfortunately have to remove the `current` dirs somehow...